src/test/java/net/glowstone/net/handler/play/player/UseItemHandlerTest.java

```diff
@@ -0,0 +1,116 @@
+package net.glowstone.net.handler.play.player;
+
+import net.glowstone.EventFactory;
+import net.glowstone.GlowServer;
+import net.glowstone.GlowWorld;
+import net.glowstone.block.GlowBlock;
+import net.glowstone.block.itemtype.ItemTypeTest;
+import net.glowstone.entity.EntityIdManager;
+import net.glowstone.entity.EntityManager;
+import net.glowstone.entity.GlowPlayer;
+import net.glowstone.entity.objects.GlowBoat;
+import net.glowstone.inventory.GlowPlayerInventory;
+import net.glowstone.net.GlowSession;
+import net.glowstone.net.message.play.player.UseItemMessage;
+import org.bukkit.GameMode;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.BlockFace;
+import org.bukkit.entity.Boat;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.Set;
+
+public class UseItemHandlerTest extends ItemTypeTest {
+
+    private EventFactory actualEventFactory;
+
+    private GlowWorld world;
+    private GlowServer server;
+    private EntityManager entityManager;
+    private EntityIdManager entityIdManager;
+    private GlowSession session;
+    private GlowPlayer player;
+    private GlowBlock emptyBlock;
+    private GlowBlock targetBlock;
+    private EventFactory eventFactory;
+
+    @Before
+    public void setupMocks() {
+        // cache the event factory
```

Comment:
I'm pretty sure some of this is redundant to the superclass's `@Before` method.

Reply:
Thank you for pointing that out.  I tried to use the serverShim but was unable to, it mocks World and the test I wrote needs GlowWorld.

I have removed the unused superclass.